### DISCUSSION TITLE
Failed to launch the Weasis because of bundle errors.

### DISCUSSION THREAD

**Discussion by generalhyang** [nroduit/Weasis#484] [General]
Hello,

I encountered some bundle errors while attempting to launch Weasis. I've attached the log containing these errors for your reference: [log of bundle errors.txt]

(https://github.com/nroduit/Weasis/files/13272011/log.of.bundle.errors.txt).

I used the release code package for version 4.2.0, and I followed the guidelines provided [here](https://weasis.org/en/getting-started/guidelines/index.html) to successfully compile and install Weasis. After adding a launcher, I was able to launch Weasis without any issues.

However, after integrating my customized plugin into Weasis version 4.2.0, I encountered bundle errors when trying to launch Weasis. Despite my efforts to resolve the issue, I couldn't find a solution. I would greatly appreciate any suggestions or guidance to help me resolve this problem.

Best regards,
Hao
[log of bundle errors.txt](https://github.com/nroduit/Weasis/files/13272061/log.of.bundle.errors.txt)

**Comment by nroduit** [comment]
If you make changes in the existing code that could be useful for everyone, then you can propose them as a pull request. Otherwise, if it is a particular development then it is better to use the plugin API (https://nroduit.github.io/en/basics/customize/build-plugins/) in order to be able to follow Weasis evolutions much more easily. In particular with the future weasis-manager server component for plugin distribution.

For OSGI dependency see https://github.com/nroduit/Weasis/issues/473
